### ISSUE TITLE
Remaining questions

### ISSUE THREAD

**Issue by seesharper** [seesharper/LightInject.NServiceBus#1]
- [ ]  Although all the tests current are passing, there is a question whether the integration should rely on scopes being managed per logical thread of execution (aka CallContext /AsyncLocal<T>) or simply per thread. The tests does not explicitly test for async/await and this could possibly be a huge problem.
- [ ] Another issue is if this integration should target 5.x or the upcoming 6.x release.

**Comment by danielmarbach** [comment]
@seesharper we are currently spiking to replace our ILMerge autofac container with lightinject, this is the PR https://github.com/Particular/NServiceBus/pull/4825. It passes all the acceptance tests and the component tests. We are very likely pulling this in. Would you want us to send in a PR with the code changes so that this can become an official NServiceBus extension?

**Comment by seesharper** [comment]
Hi @danielmarbach 

What we are talking about here is making LightInject the default container in NServiceBus, right?
If that is the case, we need to decide where this extension should live and be maintained. 
What you are suggesting is a PR so that the extension lives on in this repo and that we publish a NuGet package (LightInject.NServiceBus) that will become a dependency in NServiceBus?

If we should discover issues with the extension, would it be okay for the NServiceBus maintainers to PR into this repo?

Just clarifying :)

**Comment by danielmarbach** [comment]
Not exactly. Sorry I should have been clearer. LightInject sources will be used in NserviceBus and be completely internal. I'm talking about if you are interested to get a PR from us containing that code so that lightinject can be used as an external container by our users. Clearer?

**Comment by seesharper** [comment]
Got it. A PR would be great 👍 
BTW, honoured to see LightInject being integrated into NServiceBus 👍

**Comment by danielmarbach** [comment]
PR submitted https://github.com/seesharper/LightInject.NServiceBus/pull/2
